### ISSUE TITLE
UDM Pro requires fwmark rules on table 201

### ISSUE THREAD

**Issue by jtcressy** [tailscale/tailscale#4038]
### What is the issue?

I happen to be running tailscale on a device that puts the default route on another table (table 201) instead of putting it on the main table. Specifically, i'm running on a unifi UDM pro.
If I manually re-add the default route to the main table, tailscaled is able to connect to the control servers just fine.

I'm on version 1.20.4

Below is some relevant log output:
```
2022/02/27 19:45:28 trying bootstrapDNS("derp8d.tailscale.com", "178.62.44.132") for "controlplane.tailscale.com" ...
2022/02/27 19:45:28 bootstrapDNS("derp8d.tailscale.com", "178.62.44.132") for "controlplane.tailscale.com" error: Get "https://derp8d.tailscale.com/bootstrap-dns?q=controlplane.tailscale.com": dial tcp 178.62.44.132:443: connect: network is unreachable
2022/02/27 19:45:28 trying bootstrapDNS("derp5.tailscale.com", "2001:19f0:5801:10b7:5400:2ff:feaa:284c") for "controlplane.tailscale.com" ...
2022/02/27 19:45:28 bootstrapDNS("derp5.tailscale.com", "2001:19f0:5801:10b7:5400:2ff:feaa:284c") for "controlplane.tailscale.com" error: Get "https://derp5.tailscale.com/bootstrap-dns?q=controlplane.tailscale.com": dial tcp [2001:19f0:5801:10b7:5400:2ff:feaa:284c]:443: connect: network is unreachable
2022/02/27 19:45:28 trying bootstrapDNS("derp6.tailscale.com", "68.183.90.120") for "controlplane.tailscale.com" ...
2022/02/27 19:45:28 bootstrapDNS("derp6.tailscale.com", "68.183.90.120") for "controlplane.tailscale.com" error: Get "https://derp6.tailscale.com/bootstrap-dns?q=controlplane.tailscale.com": dial tcp 68.183.90.120:443: connect: network is unreachable
2022/02/27 19:45:28 trying bootstrapDNS("derp8b.tailscale.com", "2a03:b0c0:1:d0::ec1:e001") for "controlplane.tailscale.com" ...
2022/02/27 19:45:28 bootstrapDNS("derp8b.tailscale.com", "2a03:b0c0:1:d0::ec1:e001") for "controlplane.tailscale.com" error: Get "https://derp8b.tailscale.com/bootstrap-dns?q=controlplane.tailscale.com": dial tcp [2a03:b0c0:1:d0::ec1:e001]:443: connect: network is unreachable
2022/02/27 19:45:28 Received error: fetch control key: Get "https://controlplane.tailscale.com/key": dial tcp 52.29.8.43:443: connect: network is unreachable
2022/02/27 19:45:28 control: authRoutine: backoff: 15144 msec
```

### Steps to reproduce

remove default route, place it on another adjacent table, and verify other services (like ping) can still reach the internet but tailscaled should be constantly unable to reach any control servers with the message "network is unreachable". 

### OS

Linux

### OS version

kernel 4.19.152-al-linux-v10.2.0-v1.11.0.3921-f2e3fac, aarch64, unifi-os 1.11.0

### Tailscale version

1.20.4

### Bug report

BUG-c989981553e78e7a212129e04c3ed835b8c95a422272bd264c5ce12f3940c8a8-20220227200449Z-e441c8451363aa7d

**Comment by bradfitz** [comment]
It's also possible it's not about the table number but about which interface tailscaled binds on to hit the control plane.

Looking at your logs, tailscaled is picking eth8 but there are also a bunch of bridges:

```
2022-02-20 18:38:55.900087886 +0000 UTC: LinkChange: major, rebinding. New state: interfaces.State{defaultRoute=eth8 ifs={br0:[192.168.20.1/24 2603:xxx::1/64] br11:[192.168.1.2/24] br22:[192.168.22.1/24] br3:[192.168.3.1/24] br5:[192.168.5.1/24] br666:[192.168.66.1/24] br8:[192.168.8.1/24] eth11:[100.98.252.194/32 fd7a:115c:a1e0:efe3::6462:fcc2/128] eth8:[4n.xxx/24 2605:xxx/128]} v4=true v6=true}
2022-02-20 19:03:42.189845338 +0000 UTC: monitor: [unexpected] network state changed, but stringification didn't: xxxx
```

But eth8 does appear to be the interface with the public IP.

I thought other people already used UDM Pros, though.

Could you share here (or email us) your ip rules, iptables, interfaces, etc?

**Comment by jtcressy** [comment]
Below is output of several commands, and I've redacted any public subnets for my protection.
eth8 is my WAN connection and is _not_ a member of any bridge.
```
# ip route show
[REDACTED]/24 dev eth8 proto kernel scope link src [REDACTED - eth8 interface public ip]
192.168.1.0/24 dev br11 proto kernel scope link src 192.168.1.2
192.168.3.0/24 dev br3 proto kernel scope link src 192.168.3.1
192.168.5.0/24 dev br5 proto kernel scope link src 192.168.5.1
192.168.8.0/24 dev br8 proto kernel scope link src 192.168.8.1
192.168.20.0/24 dev br0 proto kernel scope link src 192.168.20.1
192.168.22.0/24 dev br22 proto kernel scope link src 192.168.22.1
192.168.66.0/24 dev br666 proto kernel scope link src 192.168.66.1
# ip route show table 201
default via [REDACTED - WAN next-hop] dev eth8 proto dhcp
```
ip interfaces:
```
# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 76:ac:b9:14:21:b5 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: dummy0: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN group default qlen 1000
    link/ether 06:5c:a8:43:c9:50 brd ff:ff:ff:ff:ff:ff
3: eth9: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc mq state DOWN group default qlen 10000
    link/ether 74:ac:b9:14:21:be brd ff:ff:ff:ff:ff:ff
4: eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:bd brd ff:ff:ff:ff:ff:ff
    inet [REDACTED]/24 scope global dynamic eth8
       valid_lft 77994sec preferred_lft 77994sec
    inet6 [REDACTED]/128 scope global dynamic
       valid_lft 303506sec preferred_lft 303506sec
    inet6 fe80::76ac:b9ff:fe14:21bd/64 scope link
       valid_lft forever preferred_lft forever
5: eth10: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq master br0 state UP group default qlen 10000
    link/ether 74:ac:b9:14:21:bf brd ff:ff:ff:ff:ff:ff
    inet6 fe80::76ac:b9ff:fe14:21bf/64 scope link
       valid_lft forever preferred_lft forever
6: switch0: <BROADCAST,MULTICAST,PROMISC,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 76:ac:b9:14:21:b6 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::74ac:b9ff:fe14:21b6/64 scope link
       valid_lft forever preferred_lft forever
7: gre0@NONE: <NOARP> mtu 1476 qdisc noop state DOWN group default qlen 1000
    link/gre 0.0.0.0 brd 0.0.0.0
8: gretap0@NONE: <BROADCAST,MULTICAST> mtu 1462 qdisc noop state DOWN group default qlen 1000
    link/ether 00:00:00:00:00:00 brd ff:ff:ff:ff:ff:ff
9: erspan0@NONE: <BROADCAST,MULTICAST> mtu 1450 qdisc noop state DOWN group default qlen 1000
    link/ether 00:00:00:00:00:00 brd ff:ff:ff:ff:ff:ff
10: ip_vti0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
    link/ipip 0.0.0.0 brd 0.0.0.0
11: sit0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
    link/sit 0.0.0.0 brd 0.0.0.0
12: br0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 76:ac:b9:14:21:b6 brd ff:ff:ff:ff:ff:ff
    inet 192.168.20.1/24 scope global br0
       valid_lft forever preferred_lft forever
    inet6 2603:8081:1800:e0de::1/64 scope global dynamic
       valid_lft 303506sec preferred_lft 303506sec
    inet6 fe80::8490:b9ff:fe12:6380/64 scope link
       valid_lft forever preferred_lft forever
13: eth0@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:b5 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::76ac:b9ff:fe14:21b5/64 scope link
       valid_lft forever preferred_lft forever
14: eth1@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:b6 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::76ac:b9ff:fe14:21b6/64 scope link
       valid_lft forever preferred_lft forever
15: eth2@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:b7 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::76ac:b9ff:fe14:21b7/64 scope link
       valid_lft forever preferred_lft forever
16: eth3@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:b8 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::76ac:b9ff:fe14:21b8/64 scope link
       valid_lft forever preferred_lft forever
17: eth4@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:b9 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::76ac:b9ff:fe14:21b9/64 scope link
       valid_lft forever preferred_lft forever
18: eth5@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:ba brd ff:ff:ff:ff:ff:ff
    inet6 fe80::76ac:b9ff:fe14:21ba/64 scope link
       valid_lft forever preferred_lft forever
19: eth6@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:bb brd ff:ff:ff:ff:ff:ff
    inet6 fe80::76ac:b9ff:fe14:21bb/64 scope link
       valid_lft forever preferred_lft forever
20: eth7@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:bc brd ff:ff:ff:ff:ff:ff
    inet6 fe80::76ac:b9ff:fe14:21bc/64 scope link
       valid_lft forever preferred_lft forever
21: ifb0: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN group default qlen 32
    link/ether b6:26:1a:f6:99:be brd ff:ff:ff:ff:ff:ff
22: ifb1: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN group default qlen 32
    link/ether c2:5c:51:1b:95:0b brd ff:ff:ff:ff:ff:ff
42: switch0.1@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br0 state UP group default qlen 1000
    link/ether 76:ac:b9:14:21:b6 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::74ac:b9ff:fe14:21b6/64 scope link
       valid_lft forever preferred_lft forever
61: eth10.3@eth10: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br3 state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:bf brd ff:ff:ff:ff:ff:ff
    inet6 fe80::76ac:b9ff:fe14:21bf/64 scope link
       valid_lft forever preferred_lft forever
62: eth10.5@eth10: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br5 state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:bf brd ff:ff:ff:ff:ff:ff
    inet6 fe80::76ac:b9ff:fe14:21bf/64 scope link
       valid_lft forever preferred_lft forever
63: eth10.22@eth10: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br22 state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:bf brd ff:ff:ff:ff:ff:ff
    inet6 fe80::76ac:b9ff:fe14:21bf/64 scope link
       valid_lft forever preferred_lft forever
64: eth10.8@eth10: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br8 state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:bf brd ff:ff:ff:ff:ff:ff
    inet6 fe80::76ac:b9ff:fe14:21bf/64 scope link
       valid_lft forever preferred_lft forever
65: eth10.666@eth10: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br666 state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:bf brd ff:ff:ff:ff:ff:ff
    inet6 fe80::76ac:b9ff:fe14:21bf/64 scope link
       valid_lft forever preferred_lft forever
66: eth10.11@eth10: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br11 state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:bf brd ff:ff:ff:ff:ff:ff
    inet6 fe80::76ac:b9ff:fe14:21bf/64 scope link
       valid_lft forever preferred_lft forever
67: switch0.3@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br3 state UP group default qlen 1000
    link/ether 76:ac:b9:14:21:b6 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::74ac:b9ff:fe14:21b6/64 scope link
       valid_lft forever preferred_lft forever
68: switch0.5@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br5 state UP group default qlen 1000
    link/ether 76:ac:b9:14:21:b6 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::74ac:b9ff:fe14:21b6/64 scope link
       valid_lft forever preferred_lft forever
69: switch0.22@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br22 state UP group default qlen 1000
    link/ether 76:ac:b9:14:21:b6 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::74ac:b9ff:fe14:21b6/64 scope link
       valid_lft forever preferred_lft forever
70: switch0.8@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br8 state UP group default qlen 1000
    link/ether 76:ac:b9:14:21:b6 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::74ac:b9ff:fe14:21b6/64 scope link
       valid_lft forever preferred_lft forever
71: switch0.666@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br666 state UP group default qlen 1000
    link/ether 76:ac:b9:14:21:b6 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::74ac:b9ff:fe14:21b6/64 scope link
       valid_lft forever preferred_lft forever
72: switch0.11@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br11 state UP group default qlen 1000
    link/ether 76:ac:b9:14:21:b6 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::74ac:b9ff:fe14:21b6/64 scope link
       valid_lft forever preferred_lft forever
73: br3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:bf brd ff:ff:ff:ff:ff:ff
    inet 192.168.3.1/24 scope global br3
       valid_lft forever preferred_lft forever
    inet6 fe80::64ee:4ff:fe56:d7b3/64 scope link
       valid_lft forever preferred_lft forever
74: br5: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:bf brd ff:ff:ff:ff:ff:ff
    inet 192.168.5.1/24 scope global br5
       valid_lft forever preferred_lft forever
    inet6 fe80::8444:7eff:fef4:4380/64 scope link
       valid_lft forever preferred_lft forever
75: br22: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:bf brd ff:ff:ff:ff:ff:ff
    inet 192.168.22.1/24 scope global br22
       valid_lft forever preferred_lft forever
    inet6 fe80::88a8:11ff:fed7:d95c/64 scope link
       valid_lft forever preferred_lft forever
76: br8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:bf brd ff:ff:ff:ff:ff:ff
    inet 192.168.8.1/24 scope global br8
       valid_lft forever preferred_lft forever
    inet6 fe80::fc98:a3ff:fef7:b0b/64 scope link
       valid_lft forever preferred_lft forever
77: br666: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:bf brd ff:ff:ff:ff:ff:ff
    inet 192.168.66.1/24 scope global br666
       valid_lft forever preferred_lft forever
    inet6 fe80::20ba:fff:fe68:585b/64 scope link
       valid_lft forever preferred_lft forever
78: br11: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 74:ac:b9:14:21:bf brd ff:ff:ff:ff:ff:ff
    inet 192.168.1.2/24 scope global br11
       valid_lft forever preferred_lft forever
    inet6 fe80::ec9a:2eff:fe1e:9b64/64 scope link
       valid_lft forever preferred_lft forever
79: cni0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN group default qlen 1000
    link/ether 66:a1:f6:c5:33:cc brd ff:ff:ff:ff:ff:ff
80: eth11: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1280 qdisc pfifo_fast state UNKNOWN group default qlen 500
    link/none
    inet6 fe80::414a:f8d9:4317:75ae/64 scope link stable-privacy
       valid_lft forever preferred_lft forever
```

**Comment by jtcressy** [comment]
One more comment to add:
It seems that the method tailscale uses to determine default route first attempts to use /proc/net/route and maybe resorts to using "ip route show table0" later?

https://github.com/tailscale/tailscale/blob/v1.20.4/net/interfaces/interfaces_linux.go#L188

Based on this line, what would be the behavior when it encounters a /proc/net/route output such as this?
(the code expects 00000000 for destination _and_ a mask of 00000000 on a single line to find the default route)
```
# cat /proc/net/route
Iface	Destination	Gateway 	Flags	RefCnt	Use	Metric	Mask		MTU	Window	IRTT
eth8	00FEEA2F	00000000	0001	0	0	0	00FFFFFF	0	0	0
br11	0001A8C0	00000000	0001	0	0	0	00FFFFFF	0	0	0
br3	0003A8C0	00000000	0001	0	0	0	00FFFFFF	0	0	0
br5	0005A8C0	00000000	0001	0	0	0	00FFFFFF	0	0	0
br8	0008A8C0	00000000	0001	0	0	0	00FFFFFF	0	0	0
br0	0014A8C0	00000000	0001	0	0	0	00FFFFFF	0	0	0
br22	0016A8C0	00000000	0001	0	0	0	00FFFFFF	0	0	0
br666	0042A8C0	00000000	0001	0	0	0	00FFFFFF	0	0	0
```
Further, when `ip route show table 0` is ran, it shows the default route (`default via x.x.x.x`), so ideally it should eventually use the ip command to find the default route? however it looks like that's only if tailscale is running on android based on the logic here on this line: https://github.com/tailscale/tailscale/blob/8e32002cf32c03682a5156f2f9020b088a5dde7f/net/interfaces/interfaces_linux.go#L143

If no route is found via /proc/net/route then I suppose we would expect an error to bubble its way back up the call stack but mysteriously tailscale is chugging along anyway? maybe I missed something in the logs, i'm not sure.

**Comment by jtcressy** [comment]
Ok, I found a bit of log on my side that confirmed my suspicions:
```
2022/02/28 19:22:39 link state: interfaces.State{defaultRoute= ifs={br0:[192.168.20.1/24 2603:8081:1800:e0de::1/64] br11:
[192.168.1.2/24] br22:[192.168.22.1/24] br3:[192.168.3.1/24] br5:[192.168.5.1/24] br666:[192.168.66.1/24] br8:[192.168.8.
1/24] eth8:[[REDACTED]/24 [REDACTED]/128]} v4=true v6=true}
```
It SHOULD say `defaultRoute=eth8` but instead it says `defaultRoute=` which indicates that tailscale is silently failing to find the default route, because /proc/net/route does NOT contain the default route from table 201, and it does NOT resort to using the `ip` command to fetch the default route via `ip route show table 0`

I'm fairly confident this is the root cause @bradfitz

**Comment by bradfitz** [comment]
Thanks for debugging! We don't want to start using the `ip` command in general (we've been migrating away from it and other such tools and using netlink directly), but we can use netlink to get the default route.

**Comment by bradfitz** [comment]
@jtcressy, are you able to test out https://github.com/tailscale/tailscale/pull/4042 before I merge it? Or I can just merge it & build an unstable and you can try that.

**Comment by jtcressy** [comment]
@bradfitz If you publish an image tag via ghcr.io I can pull and run it on my udmp

**Comment by bradfitz** [comment]
@maisem, do unstable builds go to ghcr.io?

**Comment by jtcressy** [comment]
@bradfitz I updated to v1.22.1 on my UDM pro but we're still getting network unreachable with interfaces.State still not setting a default route.

```
# podman exec tailscaled tailscale bugreport
BUG-c989981553e78e7a212129e04c3ed835b8c95a422272bd264c5ce12f3940c8a8-20220311182225Z-6785409e79a7047f
```

also:
```
# podman logs  tailscaled | head
2022/03/11 18:20:21 logtail started
2022/03/11 18:20:21 Program starting: v1.22.1-tc8fb4f8c7, Go 1.17.8-tsdce70b6d32: []string{"tailscaled"}
2022/03/11 18:20:21 LogID: c989981553e78e7a212129e04c3ed835b8c95a422272bd264c5ce12f3940c8a8
2022/03/11 18:20:21 logpolicy: using system state directory "/var/lib/tailscale"
2022/03/11 18:20:21 wgengine.NewUserspaceEngine(tun "tailscale0") ...
2022/03/11 18:20:21 router: v6nat = true
2022/03/11 18:20:21 dns: [rc=unknown ret=direct]
2022/03/11 18:20:21 dns: using *dns.directManager
2022/03/11 18:20:21 link state: interfaces.State{defaultRoute= ifs={br0:[192.168.20.1/24 REDACTED::1/64] br11:[192.168.1.2/24] br22:[192.168.22.1/24] br3:[192.168.3.1/24] br5:[192.168.5.1/24] br5.mac:[192.168.5.1/24] br666:[192.168.66.1/24] br8:[192.168.8.1/24] eth8:[REDACTED]} v4=true v6=true}
                                                             ^ Nothin here
```

**Comment by maisem** [comment]
> do unstable builds go to ghcr.io?

Yes they do. https://ghcr.io/tailscale/tailscale

**Comment by docbobo** [comment]
And I've looked into this a little bit more: line 171 of your change above 

```golang
c, err := rtnetlink.Dial(&netlink.Config{Strict: true})
```

results in `setsockopt: protocol not available` on my UDM pro. 

Replacing the `netlink.Config` with `nil`, I get `Error: invalid route message attributes: netlink: attribute 23 is not a uint32; length: 8`

**Comment by docbobo** [comment]
Another update: apparently, tailscaled is adding the following rules:

```
5210:  from all fwmark 0x80000 lookup main 
5230:  from all fwmark 0x80000 lookup default 
5250:  from all fwmark 0x80000 unreachable
5270:  from all lookup 52 
```

Since  UDM doesn't have the default route in the default table but in 201 (and/or 202 in case if multiple WAN interfaces) that looked problematic to me. In my case, the following two commands solved the problem of not being able to connect to the control plane:

```
ip rule add from all fwmark 0x80000 lookup 201 priority 5212
ip rule add from all fwmark 0x80000 lookup 202 priority 5211
```

Once those two commands were executed, tailscaled started up correctly and I was able to ping the hosts and subnets that were added to table 52 - at least from the UDM itself. What didn't yet work was pinging those systems from different machines.

**Comment by docbobo** [comment]
And now everything works. Pinging didn't succeed because the other host wasn't configured with `--accept-routes`. So the two additions to the list of rules seem to be the only problem.

**Comment by docbobo** [comment]
Running the following script on my UDMP now. It's first looking at the table used in rule 32766 and sets up an fwmark rule as mentioned above. Afterwards, it's monitoring the `message.log` for WAN failover messages, adjusting the rule accordingly.

```bash
RULE_PRIORITY="5225"

function getDefaultTable() {
        /sbin/ip rule list priority 32766 | cut -d " " -f 4
}

echo Routing table with default route is $(getDefaultTable)
/sbin/ip rule del priority $RULE_PRIORITY
/sbin/ip rule add priority $RULE_PRIORITY from all fwmark 0x80000 lookup $(getDefaultTable)

tail -Fn 0  /var/log/messages | while read line; do
        table=`echo $line | grep -e "ubios-udapi-server: wanFailover"`
        if [[ "$table" != ""  ]]
        then
                echo Routing table with default route is $(getDefaultTable)
                /sbin/ip rule del priority $RULE_PRIORITY
                /sbin/ip rule add priority $RULE_PRIORITY from all fwmark 0x80000 lookup $(getDefaultTable)
        fi
done
```

**Comment by jtcressy** [comment]
Thank you @docbobo! My reply is a bit late, but I just setup your script on my UDM Pro today and it works perfectly! Perfect workaround for now until the issue is completely fixed.

**Comment by juniorz** [comment]
This is still a problem for me on a Ubiquiti Cloud Gateway Ultra (UnifiOS version 4.1.13, tailscale 1.78.1 via arm64 Debian package).

`tailscaled` adds the following rules:

```
$ ip rule
[...]
5210:	from all fwmark 0x80000/0xff0000 lookup main
5230:	from all fwmark 0x80000/0xff0000 lookup default
5250:	from all fwmark 0x80000/0xff0000 unreachable
[...]
```

which breaks its startup until I manually add:

```
ip rule add from all fwmark 0x80000 lookup 201 priority 5229
```

Likely related to #11792 and #14213 (for reference).

**Comment by juniorz** [comment]
Oh, it seems to be fixed in #14452. I can confirm it works by using (unstable) tailscale 1.79.130
